dali/core/mm/cuda_vm_resource_test.cc

```diff
@@ -212,6 +212,49 @@ class VMResourceTest : public ::testing::Test {
     EXPECT_EQ(region.available_blocks, 3);
   }
 
+  void TestExceptionSafety() {
+    size_t block_size = 64_uz << 20;
+    size_t free = 0, total = 0;
+    CUDA_CALL(cudaFree(nullptr));  // initialize the default context
+    CUDA_CALL(cuMemGetInfo(&free, &total));
+    size_t va_size = next_pow2(total * 3 / 2);
```

Comment:
maybe you could add a comment here explaining the meaning of those sizes, why it throws or doesn't when we allocate this or that. not a must, just a thought